File: src/mod3_oop/lesson37_interface/OOPDemo.java
package mod3_oop.lesson37_interface;

public class OOPDemo {
    public static void main(String[] args) {
        Person kostya = Student.getInstance("Kostya Galus", 45, 3);
        Person ivan = Student.getInstance("Ivan Galustyan", 39, 2);
        Person kostya2 = Student.getInstance("Kostya Galus", 45, 3);

        System.out.println(kostya.toString());
        System.out.println(kostya.equals(ivan));
        System.out.println(kostya.hashCode());;
        System.out.println(ivan.hashCode());

        System.out.println(kostya);
        System.out.println(kostya2);
        System.out.println(kostya.equals(kostya2));
        System.out.println(kostya.hashCode());
        System.out.println(kostya2.hashCode());
    }
}

File: src/mod3_oop/lesson37_interface/OOPDemo.kt
package mod3_oop.lesson37_interface

fun main() {
    var harry = StudentKt("Harry Potter", 17, 4)
    var kostya = StudentKt("Kostya Galus", 46, 3)
    var kostya2 = StudentKt("Kostya Galus", 46, 3)

    println(harry)
    println(kostya)
    println(kostya2)

    println(harry.hashCode())
    println(kostya.hashCode())
    println(kostya2.hashCode())

    println(harry.equals(kostya2))
    println(kostya.equals(kostya2))

    // Data Class

    val ivan = StudentDataClass("Ivan Galustyan", 39, 3)
    println(ivan)
}